tfx/examples/imdb/imdb_utils_native_keras.py

```diff
@@ -27,44 +27,50 @@
 
 from tfx.components.trainer.executor import TrainerFnArgs
 
+""" 
+There are 50,000 entries in the imdb dataset. ExampleGen splits the dataset
+with a 2:1 train-eval ratio.
+Batch_size is an empirically sound configuration
+"""
 _TRAIN_BATCH_SIZE = 64 
 _TRAIN_DATA_SIZE = int(50000 * 2 / 3) 
 _EVAL_BATCH_SIZE = 64
 _EVAL_DATA_SIZE = int(50000 / 3)
 _LABEL_KEY = "sentiment"
-DELIMITERS = '.,!?() '
+_DELIMITERS = '.,!?() '
 _MAX_FEATURES = 8000
 _MAX_LEN = 100
+_LEARNING_RATE = 1e-4
+_TRAIN_EPOCHS = 10
+_DROPOUT_RATE = 0.2
+_HIDDEN_UNITS = 64
 
 def _gzip_reader_fn(filenames):
   """Small utility returning a record reader that can read gzip'ed files."""
   return tf.data.TFRecordDataset(filenames, compression_type='GZIP')
 
-def _sentiment_to_int(sentiment):
-  """Cast the label type to int64"""
-  ints = tf.cast(sentiment, tf.int64)
-  return ints
-
 def _tokenize_review(review):
   """Tokenize the reivews by spliting the reviews, then constructing a
   vocabulary. Map the words to their frequency index in the vocabulary"""
   review_sparse = tf.compat.v1.string_split(
       tf.reshape(review, shape=[-1]),
-      DELIMITERS)
+      _DELIMITERS)
 
   review_indices = tft.compute_and_apply_vocabulary(
       review_sparse,
-      default_value=_MAX_FEATURES,
+      default_value=-1,
       top_k=_MAX_FEATURES)
-
-  dense = tf.sparse.to_dense(review_indices, default_value=_MAX_FEATURES)
+  dense = tf.sparse.to_dense(review_indices, default_value=-1)
   padding_config = [[0, 0], [0, _MAX_LEN]]
   dense = tf.pad(
       dense, padding_config,
       'CONSTANT',
-      constant_values=_MAX_FEATURES)
+      constant_values=-1)
 
   padded = tf.slice(dense, [0, 0], [-1, _MAX_LEN])
```

Comment:
Lets try a different approach, lets pad in the trainer. We could go ahead and remove this then.

Reply:
There is a problem. I'm not hundred percent sure, but transform requires the output to be fixedlenfeature.
yet, the split and apply_vocab output variablelenFeature.

`The file path to a python module file, from which the 'preprocessing_fn' function will be loaded. The function must have the following signature.
def preprocessing_fn(inputs: Dict[Text, Any]) -> Dict[Text, Any]: ...

where the values of input and returned Dict are either tf.Tensor or tf.SparseTensor. Exactly one of 'module_file' or 'preprocessing_fn' must be supplied.`

From the transformer doc